cfgmgr/nbrmgr.cpp

```diff
@@ -0,0 +1,210 @@
+#include <arpa/inet.h>
+#include <netinet/in.h>
+#include <net/if.h>
+#include <unistd.h>
+
+#include "logger.h"
+#include "tokenize.h"
+#include "ipprefix.h"
+#include "macaddress.h"
+#include "nbrmgr.h"
+#include "exec.h"
+#include "shellcmd.h"
+
+using namespace swss;
+
+#define VLAN_PREFIX "Vlan"
+#define LAG_PREFIX  "PortChannel"
+
+static bool send_message(struct nl_msg *msg)
+{
+    struct nl_sock *sk = nl_socket_alloc();
+    if (!sk)
+    {
+        SWSS_LOG_ERROR("Netlink socket alloc failed");
+        return false;
+    }
+
+    nl_connect(sk, NETLINK_ROUTE);
+    bool rc = true;
+
+    if (nl_send_auto(sk, msg) < 0)
+    {
+        SWSS_LOG_ERROR("Netlink send message failed");
```

Comment:
Can you please use nl_geterror to get netlink error text? Can you please add the decrypting of nl error message on every nl library call?